### PR TITLE
migrate amazondashbutton to bnd

### PR DESCRIPTION
**Pull request by J-N-K**
This is WIP until we discussed how we provide 3rd-party libraries (https://github.com/openhab/openhab2-addons/pull/5117#issuecomment-473683175). The needed libraries (pcap4j) are available in jcenter, so compiling works. They are not available in the feature check right now.

Signed-off-by: Jan N. Klug <jan.n.klug@rub.de>

### CODE REVIEW

**Comment by J-N-K** [comment]
After digging deeper into the problem (while migrating other bindings): the problem is that the pcap4j are not OSGI bundles (`Unsupported 'Bundle-ManifestVersion' value: 1`). We could either wrap it in another bundle and deploy that or include it in the "library"-type.

**Comment by wborn** [comment]
There's also the PAX URL Wrap protocol  (using `wrap:mvn:..` instead of `mvn:..`) that might further simplify using non-bundle JARs. I haven't used it myself but it seems promising see this [page](https://davidvaleri.wordpress.com/2010/08/13/combining-karaf-features-pax-url-and-maven-to-deploy-a-plain-jar-as-an-osgi-bundle/):

```
<features>
  <feature name="myFeature" version="1.0">
    <bundle>mvn:valeri.blog.examples/karaf-wrap/1.0</bundle>
    <bundle>mvn:commons-codec/commons-codec/1.4</bundle>
    <bundle>wrap:mvn:commons-beanutils/commons-beanutils-core/1.8.3</bundle>
  </feature>
</features>
```

**Comment by J-N-K** [comment]
Nice. Works perfectly

**Comment by wborn** [comment]
Great! It should also be possible to wrap files on disk and provide instructions on what packages are imported/exported as well as providing a bundle symbolic name. :smile: 

See: https://ops4j1.jira.com/wiki/spaces/paxurl/pages/3833898/Wrap+Protocol

**Comment by J-N-K** [comment]
Maybe I can make the SOAP bindings a little bit more slim with that.

**Comment by wborn** [comment]
It might also make sense to create standalone features for these wrapped libraries that are no longer part of Java 9+ . That way they can be easily reused similar to Gson and the Jetty HTTP Client.  See this [features.xml](https://github.com/openhab/openhab-core/blob/master/features/karaf/openhab-tp/src/main/feature/feature.xml#L55).

**Comment by davidgraeff** [comment]
Guys, can you amend the  1st post of the sticky issue step by step guide with those information so that nothing is lost?

I will rewrite that guide after the migration has finished to be a generic "How to write a maven/bnd openHAB binding" and all those information come in handy. I never liked that everything was hidden in an oomph setup and a standard eclipse installation could not be used for development. That has changed now.

**Comment by J-N-K** [comment]
@davidgraeff, I can't do that.

**Comment by J-N-K** [comment]
@wborn, can we have pom-bundles, too? It would be nice if we could just depend on `openhab-jaxb` instead of the single bundles.

**Comment by wborn** [comment]
I haven't seen such a concept. But with Karaf you can also have feature dependencies. So if you add the jaxb bundles to one feature, you can just depend on that feature to get multiple bundles installed. Similar to how all add-ons depend on: `<feature>openhab-runtime-base</feature>`

**Comment by davidgraeff** [comment]
I think what J-N-K means is a bom for all bundles that were once in the java jdk and are now separate libraries

File: features/karaf/openhab-addons/src/main/feature/feature.xml [hunk 1]
```diff
@@ -37,7 +37,11 @@
 
     <feature name="openhab-binding-amazondashbutton" description="Amazon Dash Button Binding" version="${project.version}">
         <feature>openhab-runtime-base</feature>
-        <bundle start-level="80">mvn:org.openhab.binding/org.openhab.binding.amazondashbutton/${project.version}</bundle>
+        <bundle dependency="true">mvn:com.google.guava/guava/21.0</bundle>
+        <bundle dependency="true">mvn:net.java.dev.jna/jna/4.2.1</bundle>
+        <bundle dependency="true">wrap:mvn:org.pcap4j/pcap4j-core/1.6.6</bundle>
+        <bundle dependency="true">wrap:mvn:org.pcap4j/pcap4j-packetfactory-static/1.6.6</bundle>
```
Comment (wborn) [inline features/karaf/openhab-addons/src/main/feature/feature.xml]: After making a local distro build I found out that we need to add the following prerequisite to features in order to be able to use the wrap protocol in features:

`<feature prerequisite="true">wrap</feature>`

We might want to move that to the openhab-runtime-base feature so we don't have to duplicate it in many add-on features.

Also if the bundle name/symbolic name/version are not added to the wrap URL, these wrapped JARs show up very ugly with `bundle:list`:

```
220 │ Active │  80 │ 0                     │ wrap_file__home_wouter_software_openhab_instance_userdata_tmp_kar_openhab-addons-2.5.0-SNAPSHOT_com_google_api_api-common_1.7.0_api-common-1.7.0.jar
221 │ Active │  80 │ 0                     │ wrap_file__home_wouter_software_openhab_instance_userdata_tmp_kar_openhab-addons-2.5.0-SNAPSHOT_com_google_api_gax_1.42.0_gax-1.42.0.jar
222 │ Active │  80 │ 0                     │ wrap_file__home_wouter_software_openhab_instance_userdata_tmp_kar_openhab-addons-2.5.0-SNAPSHOT_com_google_auth_google-auth-library-credentials_0.13.0_google-auth-library-credentials-0.13.0.jar
223 │ Active │  80 │ 0                     │ wrap_file__home_wouter_software_openhab_instance_userdata_tmp_kar_openhab-addons-2.5.0-SNAPSHOT_com_google_auth_google-auth-library-oauth2-http_0.13.0_google-auth-library-oauth2-http-0.13.0.jar
224 │ Active │  80 │ 0                     │ wrap_file__home_wouter_software_openhab_instance_userdata_tmp_kar_openhab-addons-2.5.0-SNAPSHOT_com_google_http-client_google-http-client-jackson2_1.28.0_google-http-client-jackson2-1.28.0.jar
```

Can you add some parameters for these bundle manifest entries similar to #5219?

Comment (J-N-K) [reply features/karaf/openhab-addons/src/main/feature/feature.xml]: Sure. I‘ll also check if we can get rid of guava easily.

**Pull request merged**